homeassistant/components/environment_canada/coordinator.py

```diff
@@ -5,19 +5,22 @@
 from dataclasses import dataclass
 from datetime import timedelta
 import logging
+from typing import TypeVar
 import xml.etree.ElementTree as ET
 
 from env_canada import ECAirQuality, ECRadar, ECWeather, ec_exc
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
 from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 type ECConfigEntry = ConfigEntry[ECRuntimeData]
+ECDataTypeT = TypeVar("ECDataTypeT", ECAirQuality, ECRadar, ECWeather)
```

Comment:
It seems that mypy doesn't like this `TypeVar`...

```
homeassistant/components/environment_canada/coordinator.py:69: error: "ECDataTypeT" has no attribute "update"  [attr-defined]
homeassistant/components/environment_canada/sensor.py:290: error: "ECDataTypeT" has no attribute "metadata"  [attr-defined]
homeassistant/components/environment_canada/sensor.py:314: error: "ECDataTypeT" has no attribute "metadata"  [attr-defined]
homeassistant/components/environment_canada/sensor.py:315: error: "ECDataTypeT" has no attribute "metadata"  [attr-defined]
```

Reply:
Seems like the reason was a mix of old and new TypeVar syntax. Those don't work together.